### ISSUE TITLE
Wrong overload chosen for TypeVarTuple callbacks with Literal

### ISSUE THREAD

**Issue by cdce8p** [microsoft/pyright#7562]
**Describe the bug**

Encountered an issue with TypeVarTuple callbacks and overloads. I believe pyright is choosing the wrong overload / incorrectly rejecting the first one. The function call works without overloads (see `func2`).

```py
from typing import Any, Generic, Literal, TypeVarTuple, overload

_Ts = TypeVarTuple("_Ts")

class SignalType(Generic[*_Ts]): ...
signal = SignalType[Literal["continue", "stop"]]()

@overload
def func1(signal: SignalType[*_Ts], *args: *_Ts) -> None: ...
@overload
def func1(signal: str, *args: Any) -> None: ...
def func1(signal: SignalType[*_Ts] | str, *args: *_Ts) -> None: ...

func1(signal, "continue")  # error


# This is fine
def func2(signal: SignalType[*_Ts], *args: *_Ts) -> None: ...
func2(signal, "continue")
```

```
test.py:14:1 - error: No overloads for "func1" match the provided arguments (reportCallIssue)
test.py:14:7 - error: Argument of type "SignalType[Literal['continue', 'stop']]" cannot be assigned to parameter "signal" of type "str" in function "func1"
"SignalType[Literal['continue', 'stop']]" is incompatible with "str" (reportArgumentType)
```

From VSCode: `"Overload 2 is the closest match"`

**VS Code extension or command-line**
pyright 1.1.355

**Comment by erictraut** [comment]
Thanks for filing the issue. I'm able to repro, and I agree this is a bug. I'll investigate further.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.358, which I just published.
